File: packages/android/src/main/java/com/kozmos/components/MapOverlay/MapOverlay.figma.kt
package com.kozmos.components.mapoverlay

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8429")
class KozmosMapOverlayConnect {
    @FigmaProperty(FigmaType.Text, "Overlay Title Text")
    val title: String = "Overlay"

    // Width is a web t-shirt scale; on Compose the host sizes the overlay with
    // modifiers, recorded as intentional in the variant analyzer.
    @Composable
    fun ComponentExample() {
        KozmosMapOverlay(position = OverlayPosition.TOP_LEFT) {}
    }
}

File: packages/ios/Sources/Components/MapOverlay/MapOverlay.figma.swift
import SwiftUI
import Figma

struct KozmosMapOverlayConnect: FigmaConnect {
    let component = KozmosMapOverlay<Text>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8429"

    @FigmaString("Overlay Title Text")
    var title: String = "Overlay"

    // Width is a web t-shirt scale; on native the host sizes the overlay with
    // frame modifiers, which is recorded as intentional in the variant
    // analyzer. position is renderer placement.
    var body: some View {
        KozmosMapOverlay(position: .topLeft) {
            Text(self.title)
        }
    }
}
